### DISCUSSION TITLE
Mark unused variables in Jupyter notebooks

### DISCUSSION THREAD

**Discussion by TernaryM01** [microsoft/pylance-release#3920] [Enhancement]
In Python (*.py) files, unused variables (in my case, function parameters that are never used in the function) are marked by being greyed out.

However, that is not the case for Python code cells in Jupyter notebooks (*.ipynb). I have installed Jupyter extensions, installed pylint in the environment, and searched the web. Such a feature doesn't seem to exist yet. Please add it.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by debonte** [comment]
@heejaechang, looks like your recent fix for unused imports was specific to imports. Could it be easily expanded to handle the other unused symbol scenarios? https://github.com/microsoft/pylance-release/issues/3175

**Reply by heejaechang** [reply]
we might be able to, but in notebook, in certain case such as last expression in cell, could be unused by users but used by notebook itself? so there might be more cases to consider?

**Reply by Servinjesus1** [reply]
So presumably in that case, users should have the ability to mark a variable as utilized, or otherwise deal with it.

As a workaround for now, copying all cell contents into a single function definition works for filtering unused variables...

**Comment by Servinjesus1** [comment]
This would be useful, for example for pruning gather outputs. However, I notice pylance doesn't mark global-scope variables unused either. Is this by design?
